docs/src/design/chiplets/main.md

```diff
@@ -154,6 +140,64 @@ The following constraints ensure that the chiplets are stacked correctly by rest
+
+The chiplets bus constraints are defined by the components that use it to communicate.
+
+Lookup requests are sent to the chiplets bus by the following components:
+
+- The stack sends requests for [bitwise](../stack/u32_ops.md#u32and), [memory](../stack/io_ops.md#memory-access-operations), and [cryptographic hash operations](../stack/crypto_ops.md).
+- The decoder sends requests for [hash operations](../decoder/main.md#program-block-hashing) for program block hashing.
+- The decoder sends a procedure access request to the [Kernel ROM chiplet](./kernel_rom.md) for each `SYSCALL` during [program block hashing](../decoder/main.md#program-block-hashing).
+
+Responses are provided by the [hash](./hasher.md#chiplets-bus-constraints), [bitwise](./bitwise.md#chiplets-bus-constraints), [memory](./memory.md#chiplets-bus-constraints), and [kernel ROM](./kernel_rom.md#chiplets-bus-constraints) chiplets.
+
+## Chiplets virtual table
+
+Some chiplets require the use of a [virtual table](../multiset.md#virtual-tables) to maintain and enforce the correctness of their internal state. Because the length of these virtual tables does not exceed the length of the chiplets themselves, a single virtual table called $vt_{chip}$ can be shared by all chiplets.
+
+Currently, the chiplets virtual table combines two virtual tables:
+- the hash chiplet's [sibling table](./hasher.md#sibling-table-constraints)
+- the kernel ROM chiplet's [kernel procedure table](./kernel_rom.md#kernel-procedure-table-constraints)
+
+To combine these correctly, the [running product column](../multiset.md) for this table must be constrained not only at the beginning and the end of the trace, but also where the hash chiplet ends and where the kernel ROM chiplet begins. These positions can be identified using the chiplet selector columns.
+
+### Chiplets virtual table constraints
+
+The expected boundary values for each chiplet's portion of the virtual table must be enforced. This can be done as follows.
+
+For the sibling table to be properly constrained, the value of the running product column must be $1$ when the sibling table starts and finishes. This can be achieved by:
+- enforcing a boundary constraint for $vt_{chip}=1$ at the first row
+- using the the following transition constraint to enforce that the value is once again $1$ at the last cycle of the hash chiplet.
+
+> $$
+(s'_0 - s_0) \cdot (1 - vt_{chip}) = 0 \text{ | degree} = 2
+$$
+
+For the kernel procedure table to be properly constrained, the value must be $1$ when it starts, and it must be equal to the product of all of the kernel ROM procedures when it finishes. This can be achieved by:
+- enforcing a boundary constraint against the last row for the value of all of the kernel ROM procedures
+- using the following transition constraint to enforce that when the active chiplet changes to the kernel ROM chiplet the value is $1$.
+
+> $$
+s_0 \cdot s_1 \cdot (s'_2 - s_2) \cdot (1 - vt'_{chip}) = 0 \text{ | degree} = 4
+$$
```

Comment:
I'm not sure this is necessary - but let's keep it. We should probably make a note somewhere that this is something we can potentially optimize away.

Reply:
Hm. I thought about it, but to me it seems that we need to constrain the start & end values of each virtual table, and if we combine the virtual tables, then we would still need to constrain the start/end of those "chunks" of the combined table

Reply:
My thinking was that we already verify that the value in the column is `ONE` at the end of the hash chiplet segment, and no other chiplet can change the value - so, it should remain `ONE` at the start of kernel ROM chiplet. But again, I think it is fine to keep for now and we can revisit later if needed.